helix-term/src/ui/prompt.rs

```diff
@@ -59,14 +59,14 @@ impl Prompt {
     pub fn new(
         prompt: Cow<'static, str>,
         history_register: Option<char>,
-        mut completion_fn: impl FnMut(&str) -> Vec<Completion> + 'static,
+        completion_fn: impl FnMut(&str, &Context) -> Vec<Completion> + 'static,
         callback_fn: impl FnMut(&mut Context, &str, PromptEvent) + 'static,
     ) -> Self {
         Self {
             prompt,
             line: String::new(),
             cursor: 0,
-            completion: completion_fn(""),
+            completion: Vec::new(),
```

Comment:
Hmm, with this change initial completion will no longer pop up?

Reply:
Ah, indeed. Initial `:` completion breaks with this... Looks like I'll need to pass in the `Context`...

Reply:
Golly, that's gonna be such a pain... Might even be impossible, with the current structure... Passing in a `compositor::Context` means that every prompt will also need to pass in a `compositor::Context`, and these prompts are called from places without one readily available without also making *them* pass in a `compositor::Context`...

Reply:
I was able to fix this by setting the initial completion inside the `Prompt`'s `render`, but I don't know if that's the best idea, or any potential drawbacks. But it's a heck of a lot easier than trying to get a `compositor::Context` everywhere.

https://github.com/helix-editor/helix/commit/63ab472cfe968e528e1030261015ba20d1c5cd90

Reply:
Sorry for the delay here, I'm still thinking if there's a better way to do this without waiting for the call to render. A bit of a hack, but you could call it inside `required_size`? We guarantee to call that immediately when the component is inserted.